packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -205,8 +205,14 @@ abstract class FlutterCommand extends Command<void> {
       hide: hide,
     );
     argParser.addOption('web-user-data-dir',
-      help: 'The path the browser should use for its user data dir.',
-      hide: true,
+      help: 'The path the browser should use for its user data directory. '
+        'This directory stores the user\'s browser settings, bookmarks, '
+        'saved passwords, etc. If the directory doesn\'t exist, a new one '
+        'will be created with the correct structure, otherwise the data '
+        'in the existing directory will be loaded. If not provided, a temporary '
+        'one will be used.',
```

Comment:
Browser -> Chromium Browser

```suggestion
        'directory will be used instead.',
```

I would also add in a new paragraph something like:

```
using an existing data directory may cause flutter run to fail if there are instances of a Chromium browser running also using that data directory
```